File: Benchmarking/src/gen_3_30.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: s0
    var p1: Double
  }
  struct s2 {
    var p0: s1
    var p1: s1
  }
  struct s3 {
    var p0: [s0]
    var p1: [s1]
  }
  struct s8 {
    var p0: s2
    var p1: s3
  }
  @inline(never)
  func f0(_ v0: [s8], _ v1: [s1], _ v2: Double) -> Double {
    let v8: s8 = v0[0]
    let v6: s3 = v8.p1
    let v7: s1 = v1[0]
    let v10: [s1] = [v7, v7, v7, v7, v7]
    let v24: [s1] = v6.p1
    let v85: s1 = v24[0]
    let v46: s1 = v10[1]
    let v57: Double = v46.p1
    let v44: Double = v57 / v57
    let v99: Double = v85.p1
    let v151: Double = v99 - v2
    let v141: Double = v151 + v44
    return v141
  }
  func benchmark() {
    let v0: [s8] = [s8(p0: s2(p0: s1(p0: s0(p0: [[0.0], [1.0]], p1: [[2.0], [3.0]]), p1: 4.0), p1: s1(p0: s0(p0: [[5.0], [6.0]], p1: [[7.0], [8.0]]), p1: 9.0)), p1: s3(p0: [s0(p0: [[10.0], [11.0]], p1: [[12.0], [13.0]]), s0(p0: [[14.0], [15.0]], p1: [[16.0], [17.0]])], p1: [s1(p0: s0(p0: [[18.0], [19.0]], p1: [[20.0], [21.0]]), p1: 22.0), s1(p0: s0(p0: [[23.0], [24.0]], p1: [[25.0], [26.0]]), p1: 27.0)])), s8(p0: s2(p0: s1(p0: s0(p0: [[28.0], [29.0]], p1: [[30.0], [31.0]]), p1: 32.0), p1: s1(p0: s0(p0: [[33.0], [34.0]], p1: [[35.0], [36.0]]), p1: 37.0)), p1: s3(p0: [s0(p0: [[38.0], [39.0]], p1: [[40.0], [41.0]]), s0(p0: [[42.0], [43.0]], p1: [[44.0], [45.0]])], p1: [s1(p0: s0(p0: [[46.0], [47.0]], p1: [[48.0], [49.0]]), p1: 50.0), s1(p0: s0(p0: [[51.0], [52.0]], p1: [[53.0], [54.0]]), p1: 55.0)]))]
    let v1: [s1] = [s1(p0: s0(p0: [[56.0], [57.0]], p1: [[58.0], [59.0]]), p1: 60.0)]
    let v2: Double = 61.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 61.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()

File: Benchmarking/src/gen_3_30.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1): p0(v0), p1(v1) { }
  };
  struct s1 {
    s0 p0;
    double p1;
    s1(s0 v0, double v1): p0(v0), p1(v1) { }
  };
  struct s2 {
    s1 p0;
    s1 p1;
    s2(s1 v0, s1 v1): p0(v0), p1(v1) { }
  };
  struct s3 {
    std::vector<s0> p0;
    std::vector<s1> p1;
    s3(std::vector<s0> v0, std::vector<s1> v1): p0(v0), p1(v1) { }
  };
  struct s8 {
    s2 p0;
    s3 p1;
    s8(s2 v0, s3 v1): p0(v0), p1(v1) { }
  };
  __attribute__((noinline))
  double f0(const std::vector<s8> &v0, const std::vector<s1> &v1, const double &v2) {
    const s8 v8 = v0[0];
    const s3 v6 = v8.p1;
    const s1 v7 = v1[0];
    const std::vector<s1> v10 { v7, v7, v7, v7, v7 };
    const std::vector<s1> v24 = v6.p1;
    const s1 v85 = v24[0];
    const s1 v46 = v10[1];
    const double v57 = v46.p1;
    const double v44 = v57 / v57;
    const double v99 = v85.p1;
    const double v151 = v99 - v2;
    const double v141 = v151 + v44;
    return v141;
  }
  int main() {
    std::vector<s8> v0({ { { { { { { 0.0 }, { 1.0 } }, { { 2.0 }, { 3.0 } } }, 4.0 }, { { { { 5.0 }, { 6.0 } }, { { 7.0 }, { 8.0 } } }, 9.0 } }, { { { { { 10.0 }, { 11.0 } }, { { 12.0 }, { 13.0 } } }, { { { 14.0 }, { 15.0 } }, { { 16.0 }, { 17.0 } } } }, { { { { { 18.0 }, { 19.0 } }, { { 20.0 }, { 21.0 } } }, 22.0 }, { { { { 23.0 }, { 24.0 } }, { { 25.0 }, { 26.0 } } }, 27.0 } } } }, { { { { { { 28.0 }, { 29.0 } }, { { 30.0 }, { 31.0 } } }, 32.0 }, { { { { 33.0 }, { 34.0 } }, { { 35.0 }, { 36.0 } } }, 37.0 } }, { { { { { 38.0 }, { 39.0 } }, { { 40.0 }, { 41.0 } } }, { { { 42.0 }, { 43.0 } }, { { 44.0 }, { 45.0 } } } }, { { { { { 46.0 }, { 47.0 } }, { { 48.0 }, { 49.0 } } }, 50.0 }, { { { { 51.0 }, { 52.0 } }, { { 53.0 }, { 54.0 } } }, 55.0 } } } } });
    std::vector<s1> v1({ { { { { 56.0 }, { 57.0 } }, { { 58.0 }, { 59.0 } } }, 60.0 } });
    double v2(61.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1, v2);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }
